SrfGen/srfinfo2vm.py

```diff
@@ -95,10 +95,24 @@ def auto_z(mag, depth):
 
 
 # simulation time based on area
-def auto_time2(xlen, ylen, ds_multiplier):
-    rrup = max(xlen, ylen) / 2.0
-    s_wave_arrival = rrup / 3.2
-    siteprop.Rrup = rrup
+def auto_time2(vm_corners, srf_corners, ds_multiplier):
+    """Calculates the sim duration from the bounds of the vm and srf
+    :param vm_corners: A list of (lon, lat) tuples
+    :param srf_corners: A [4n*2] numpy array of (lon, lat) pairs where n is the number of planes in the srf
+    :param ds_multiplier: An integer"""
+    # S wave arrival time is determined by the distance from the srf centroid to the furthest corner
+    s_wave_arrival = (
+            max([
+                geo.ll_dist(*corner, (srf_corners[:, 0].max + srf_corners[:, 0].min) / 2,
+                            (srf_corners[:, 1].max + srf_corners[:, 1].min) / 2)
+                for corner in vm_corners
+            ]) / 3.2
+    )
```

Comment:
And maybe to simplify the whole thing (up to you though):
srf_locations = (a.max(axis=0) + a.min(axis=0)) / 2
geo.get_distance(vm_corners, srf_locations[0], srf_locations[1]).max() / 3.2

where a = srf_corners
Needs some testing though haha